### PR TITLE
Initial OSGi docs

### PR DESCRIPTION
**Pull request by neykov**

### CODE REVIEW

File: guide/misc/osgi.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+title: FAQ
+layout: website-normal
+---
+
+# Running Apache Brooklin inside Karaf container
```
Comment (sjcorbett) [inline guide/misc/osgi.md]: `Brooklin`

File: guide/misc/osgi.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    SecurityProvider implementations configured in Brooklyn falling back to auto generating
+    the password. To configure a custom JAAS realm see the jetty.xml file in "brooklyn-server/karaf/jetty-config/src/main/resources"
+    and override it by creating a custom one in "etc" folder. Point the "webconsole" login service
+    to the JAAS realm you would like to use.
+   * For other Jetty related configuration consult the Karaf and pax-web docs.
+
+
+## Caveats
+
+In the OSGi world specifying class names by string in Brooklyn's configuraiton will work only
```
Comment (sjcorbett) [inline guide/misc/osgi.md]: `configuraiton`

**Comment by sjcorbett** [comment]
Looks good. Do you want me to double check any of the instructions in particular?

File: guide/misc/osgi.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+---
+title: FAQ
```
Comment (geomacy) [inline guide/misc/osgi.md]: The title is misleading, it's not a FAQ page really - change to 'OSGI'?

File: guide/misc/osgi.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+{% highlight bash %}
+bin/karaf
+{% endhighlight %}
+
+To start in debug mode use
+
+{% highlight bash %}
+bin/karaf debug
+{% endhighlight %}
+
```
Comment (m4rkmckenna) [inline guide/misc/osgi.md]: Might be worth mentioning `karaf clean` to reset config admin / installed features to default

File: guide/misc/osgi.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
+---
+title: FAQ
+layout: website-normal
+---
+
+# Running Apache Brooklin inside Karaf container
+
+The Apache Brooklyn Karaf based distribution lives in brooklyn-server/karaf/apache-brooklyn folder.
```
Comment (geomacy) [inline guide/misc/osgi.md]: Worth adding a note on the status of the Karaf distro, with caveats about not everything necessarily working yet.

File: guide/misc/osgi.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+{% highlight bash %}
+JAVA_DEBUG_OPTS='-Xdebug -Xnoagent -Djava.compiler=NONE -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005' bin/karaf debug
+{% endhighlight %}
+
+## Configuring
+
+You can use the standard ~/.brooklyn/brooklyn.properties file to configure Brooklyn. Alternatively
+create etc/brooklyn.cfg inside the distribution folder (same file format). The keys in the former override
+those in the latter.
```
Comment (geomacy) [inline guide/misc/osgi.md]: I'd reword this to say 'use <install-dir>/etc/brooklyn.cfg' but making a note that ~/.brooklyn/brooklyn.properties is still supported for backward compatibility.   But also explain that values in brooklyn.cfg can be viewed and managed dynamically via the OSGI configuration admin commands in Karaf, e.g. config:property-set, so this is better suited for working in an OSGI environment.

File: guide/misc/osgi.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+Web console related configuration is done through the corresponding Karaf mechanisms:
```
Comment (geomacy) [inline guide/misc/osgi.md]: Need a blank line after this one if the bullets below are to render properly.

**Comment by geomacy** [comment]
What do we do about the numerous other pages where 'brooklyn.properties' is referenced?   Is it best to leave them as-is until Brooklyn moves to a Karaf-only distribution?

**Comment by geomacy** [comment]
that's all the comments from me...

**Comment by neykov** [comment]
Added a new commit addressing the comments so it's clear what the changes are. Will squash before merging.
@sjcorbett The instructions are mostly one-liners copy & pasted from my workflow, so no need to go through them. What I'd be really interested in is an "outsider" testing the distribution and reporting his experience.
